infinispan/src/main/java/org/hibernate/search/infinispan/impl/InfinispanDirectoryProvider.java

```diff
@@ -129,8 +136,30 @@ public void start(DirectoryBasedIndexManager indexManager) {
 		log.debugf( "Initialized Infinispan index: '%s'", directoryProviderName );
 	}
 
+	private void augmentDirectoryWithAsyncDeletes(org.infinispan.lucene.directory.BuildContext directoryBuildContext) {
+		Executor executor = deletesExecutor.getExecutor();
+		try {
+			Method operationsExecutor = directoryBuildContext.getClass().getMethod(
+					"deleteOperationsExecutor",
+					Executor.class
+			);
+			operationsExecutor.invoke( directoryBuildContext, executor );
+		}
+		catch (NoSuchMethodException e) {
+			log.notSettingAsyncDeletes();
+		}
+		catch (InvocationTargetException e) {
+			log.notSettingAsyncDeletes();
```

Comment:
The reason why is like that is that I'd like to avoid showing exceptions in the log. This is a best effort to enable async deletes, it's not suppose to 'scare' users  :)